jdaviz/cli.py

```diff
@@ -71,16 +70,38 @@ def main(filepaths=None, layout='default', instrument=None, browser='default',
     solara.theme = theme
     solara.jdaviz_verbosity = verbosity
     solara.jdaviz_history_verbosity = history_verbosity
-    args = []
-    if hotreload:
-        args += ['--auto-restart']
+    run_solara(host=host, port=port, theme=theme, browser=browser, production=not hotreload)
+
+
+def run_solara(host, port, theme, browser, production: bool = True):
+    os.environ["SOLARA_APP"] = "jdaviz.solara"
+    import solara.server.starlette
+    import solara.server.settings
+    solara.server.settings.theme.variant = theme
+    solara.server.settings.theme.loader = "plain"
+    solara.server.settings.main.mode = "production" if production else "development"
+
+    server = solara.server.starlette.ServerStarlette(host="localhost", port=port)
+    print(f"Starting server on {server.base_url}")
+    server.serve_threaded()
+    server.wait_until_serving()
+    if browser == "qt":
+        from qtpy.QtWidgets import QApplication
+        from qtpy.QtWebEngineWidgets import QWebEngineView
+        from qtpy import QtCore
```

Comment:
if these fail to import, should we throw an error with instructions to install the optional dependencies?

Reply:
Indeed, good idea.

Reply:
I've put this in qt.py